u2f-ref-code/java/src/com/google/u2f/server/impl/U2FServerReferenceImpl.java

```diff
@@ -151,7 +147,7 @@ public SecurityKeyData processRegistrationResponse(
         appIdSha256, clientDataSha256, keyHandle, userPublicKey);
 
     Set<X509Certificate> trustedCertificates = dataStore.getTrustedCertificates();
-    if (!trustedCertificates.contains(attestationCertificate)) {
+    if (!trustedCertificates.contains(attestationCertificateChain)) {
```

Comment:
My Java Collection foo isn't all that strong, but, how is Set<X>.contains(X[]) supposed to behave? (Does that notation make any sense?) You're passing an array of type X509Certificate to a method that expects X509Certificate. Do you mean containsAll, or do you intend to pass the root of the chain instead, or does this just magically work?